### DISCUSSION TITLE
Virtual numeric pad for Raspberry touch screen

### DISCUSSION THREAD

**Discussion by danielcqar** [frangoteam/FUXA#215] [Q&A]
Hi there,
I really like this project, I started a week ago testing its UI capabilities and I found it can covers almost all my needs but... (always there is a but)... I need to run it (server&client as a HMI on Rpi4) on a touch screen... NO physical keyboard... NO mouse...
Before fighting with raspbian or Chromium add-ons I ask the question: has FUXA any built-in support for on-screen numpad?, if not: can I build an on-screen numpad with the available controls?.
Thanks in advance.

**Comment by concept10** [comment]
You may want to consider an OSD at the distribution level. I'm sure an on screen keyboard is included with Raspbian as an example. Or possibly use some control to trigger an OSD.

**Reply by danielcqar** [reply]
Thanks, I'll take your suggestion if there is no a non-global solution.

**Comment by unocelli** [comment]
There are a nice [virtual keyboard](https://www.npmjs.com/package/@ngx-material-keyboard/core) for Angular, should be easy to integrate in the input control (only for view not in editor). I will try to add it as soon as I find the time ;)

**Reply by danielcqar** [reply]
That's what I was looking for!. 
It can be customized as a numeric-only layout. 
I'll try to integrate it (while the official version is cooked ;) )...

**Reply by afbenevides** [reply]
Hello I was looking for touch input keyboard on PI with Fuxa and found this discussion. Has that been implemented by one of you?

Thanks a lot.

**Comment by afbenevides** [comment]
Someone also pointed me out this tool that could be implemented I guess? 
https://github.com/furcan/KioskBoard
What do you think?

**Reply by afbenevides** [reply]
Just wanted to share that I finally used raspian + kiosk mode + chromium keyboard extension which for now works pretty good.

**Reply by MatthewReed303** [reply]
Thanks for update, I'm just about to go through same process. In the future built in keyboard and numpad would be great.

**Reply by afbenevides** [reply]
I now find that the virtual keyboard chromium extension does not work anymore to input values in fields. In fact we see values typed, but the web app does not seem to register them. I also get not a number error on the fields when I enter a number with the virtual keyboard. When I use the usb keyboard on same machine just seconds after using the keyboard it works. I saw same kind of behaviour in the login screen, ok would never get available if I typed with virtual keyboard, but as soon as I used the real keyboard it worked.

if you have any idea I would be great full! Do you still plan on implementing a virtual keyboard directly in the app?

thanks again
I’m on 1.1.9
Raspian + kiosk mode + chromium virtual keyboard extension not working

**Reply by afbenevides** [reply]
@unocelli please also share if you have an idea for an alternative solution that I could test thanks

**Comment by unocelli** [comment]
I think that integrate a virtual keyboard like [ngx-material-keyboard](https://www.npmjs.com/package/@ngx-material-keyboard/core) will solve the issue.

**Reply by MatthewReed303** [reply]
Here is another onscreen keyboard option:
https://github.com/simple-keyboard/angular-simple-keyboard
Here is the demo:
https://hodgef.com/simple-keyboard/demos/?d=darkTheme
Here is the code sandbox:
https://codesandbox.io/s/github/hodgef/simple-keyboard-npm-demos/tree/dark-theme/?from-embed=&file=/src/index.js:231-256
Here is num only:
https://codesandbox.io/s/github/hodgef/simple-keyboard-npm-demos/tree/numeric-theme/?from-embed

What would be required to integrate with fuxa? Maybe have a setting to enable/disable virtual keyboard?

**Comment by robbudge** [comment]
I also use touchscreens, on another SCADA system we pass the target variable to a dialog. The Dialog has a local string that buttons 0-9 and decimal each add the associated character to a local string. 
allowing the operator to write the string 123.456
On pressing the enter button the string is converted to a number and written to the passed variable.

considering touch screens are very common, someway to build or include dialog-keypad needs to be included.
This fundamental control  could be a deal breaker for most

**Comment by unocelli** [comment]
Virtual Keyboard is finally available in release [1.1.11](https://github.com/frangoteam/FUXA/releases/tag/v1.1.11)
